doc/jvm/xgboost4j_spark_tutorial.rst

```diff
@@ -153,6 +153,54 @@ Now, we have a DataFrame containing only two columns, "features" which contains
 "sepal length", "sepal width", "petal length" and "petal width" and "classIndex" which has Double-typed
 labels. A DataFrame like this (containing vector-represented features and numeric labels) can be fed to XGBoost4J-Spark's training engine directly.
 
+Dealing with missing values (NaN / Null)
+--------------------------
+
+In case a feature (column) contains missing values (NaN / NULLs) for any reason (could be related to business logic / wrong data ingestion process / etc.), one should decide on a strategy of how to handle it.
+
+Possible ways are - 
+1. One can ignore it in the VectorAssembler (using setHandleInvalid = "skip"); 
+2. keep it (using setHandleInvalid = "keep"), and then XGBoost will handle it.
+
+In case a user chose strategy #2, it should handle it in the below manner:
+
+2.1. Set some irregular value to that feature.
+For example: A feature that contains values in the range of 0 to 1, we can set the missing values as -999.
+(Note: The missing value should be replaced with the same value on whole of the dataset, and not only on a specific feature)
+
+2.2. Once we have "padded" our dataframe with irregular values for the missing values, we should tell our XGBoostClassifier / XGBoostRegressor about it, via a parameter called "missing", which represents the missing value.
+The algorithm will automatically learn what's the ideal direction to go when a value is missing, based on that value. 
+
+Example:
+
+.. code-block:: scala
+
+  import ml.dmlc.xgboost4j.scala.spark.XGBoostClassifier
+  val xgbParam = Map("eta" -> 0.1f,
+        "missing" -> -999,
+        "objective" -> "multi:softprob",
+        "num_class" -> 3,
+        "num_round" -> 100,
+        "num_workers" -> 2)
+  val xgbClassifier = new XGBoostClassifier(xgbParam).
+        setFeaturesCol("features").
+        setLabelCol("classIndex")
+
+3. Then, once we transform (test) our XGBoost model, we should see that indeed the number of records match the number we "fed" our model with.
+
+
+Dealing with 0's as missing values
+--------------------------
+In case a dataset contains 0 as a missing value, and those are meaning values (i.e. represent some kind of a behavior in the data), one should handle it, due to the fact that Spark's VectorAssembler transformer only accepts 0 as the missing values. This issue creates a problem when the user has 0 as meaningful value, plus there are enough 0's to use SparseVector.
```

Comment:
0 as `feature` value???

Reply:
yes, i.e. a value inside a feature within the dataset. A meaningful value.
Would you like me to re-phrase it better?

Reply:
In the case that a feature column contains missing values for any reason (could be related to business logic / wrong data ingestion process / etc.), the user should decide on a strategy of how to handle it. The choice of approach depends on the value representing 'missing' which fall into four categories: (1) 0; (2) NaN; (3) NULL; (4) any other value besides the earlier three cases.

We introduce the following approaches dealing with missing value and their fitting scenarios:

1. skip VectorAssembler (using setHandleInvalid = "skip") directly: used in (2), (3);
2. keep it (using setHandleInvalid = "keep"), and set `missing` parameter in XGBClassifier/XGBRegressor as the value representing `missing`: used in (2) and (4);
3. keep it (using setHandleInvalid = "keep") and transform to other irregular values: used in (3)
4. transform to other values: for case (1), blah blah, blah

Reply:
@CodingCat done, is part of updated pull request (hopefully final pull req.).